src/main/groovy/edu/oregonstate/mist/personsapi/PersonsResource.groovy

```diff
@@ -0,0 +1,127 @@
+package edu.oregonstate.mist.personsapi
+
+import com.codahale.metrics.annotation.Timed
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import edu.oregonstate.mist.personsapi.db.PersonsDAO
+import groovy.transform.TypeChecked
+
+import javax.annotation.security.PermitAll
+import javax.imageio.ImageIO
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import java.awt.image.BufferedImage
+
+@Path("persons")
+@Produces(MediaType.APPLICATION_JSON)
+@PermitAll
+@TypeChecked
+class PersonsResource extends Resource {
+    private final PersonsDAO personsDAO
+    private PersonUriBuilder personUriBuilder
+    private final Integer maxImageWidth = 2000
+
+    PersonsResource(PersonsDAO personsDAO, URI endpointUri) {
+        this.personsDAO = personsDAO
+        this.endpointUri = endpointUri
+        this.personUriBuilder = new PersonUriBuilder(endpointUri)
+    }
+
+    @Timed
+    @GET
+    Response list(@QueryParam('onid') String onid,
+                  @QueryParam('osuID') String osuID,
+                  @QueryParam('osuUID') String osuUID) {
+
+        def id = [onid, osuID, osuUID].findAll { it }
+        if (id.size() != 1) {
+            badRequest('The number of input parameter(s) is not equal to one.').build()
+        } else {
+            def persons = personsDAO.getPersons(onid, osuID, osuUID)
+            ResultObject res = new ResultObject(
+                data: persons.collect {
+                    new ResourceObject(
+                        id: it.osuID,
+                        type: 'person',
+                        attributes: it,
+                        links: ['self': personUriBuilder.personUri(it.osuID)]
+                    )
+                }
+            )
+            ok(res).build()
+        }
+    }
+
+    @Timed
+    @GET
+    @Path('{osuID: [0-9]{9}}')
```

Comment:
I'm hesitant about declaring the exact number of digits the ID should be. One side effect right now is that if I search with a 9 digit id that doesn't exist in the database, I will get a 404 with the application/json header. If I search with an id that's not 9 digits, it won't even invoke the method and the 404 it returns doesn't have the application/json header. That's an issue either way, but I think it should be fixed in a different pr for the skeleton. In the mean time, perhaps we shouldn't validate the length of the id?